xbmc/platform/linux/test/TestSysfsPath.cpp

```diff
@@ -8,24 +8,44 @@
 
 #include "platform/linux/SysfsPath.h"
 
+#include <chrono>
+#include <string>
+#include <sstream>
 #include <fstream>
 
 #include <gtest/gtest.h>
 
 struct TestSysfsPath : public ::testing::Test
 {
-  ~TestSysfsPath() { std::remove("/tmp/kodi-test"); }
+  std::string GetTestFilePath()
+  {
+    using namespace std::chrono;
 
-  std::ofstream m_output{"/tmp/kodi-test"};
+    const char *test_tmpdir = getenv("TMPDIR");
+    if (!test_tmpdir || test_tmpdir[0] == '\0') {
+      return "";
```

Comment:
we need a fallback when TMPDIR is not set, which will often be the case:
```suggestion
      test_tmpdir = "/tmp/kodi-test";
```